instancetypes: Add PreferredCPUTopology and SpreadOptions docs (#817)
kubevirt/user-guide: docs/user_workloads/instancetypes.md

    "name": "cloudinitdisk"
  }
]
```

### PreferredCPUTopology

> A preference can optionally include a `PreferredCPUTopology` that is used to apply vCPUs supplied by an instance type to the guest visible CPU topology of the `VirtualMachineInstance`.

Reviewer: ```suggestion
A preference can optionally include a `PreferredCPUTopology` that defines how the guest visible CPU topology of the `VirtualMachineInstance` is constructed from vCPUs supplied by an instance type.
```

```yaml
apiVersion: instancetype.kubevirt.io/v1beta1
kind: VirtualMachinePreference
metadata:
  name: example-preference-cpu-topology
spec:
  cpu:
    preferredCPUTopology: cores
```

The allowed values for `PreferredCPUTopology` include:

* `sockets` (default) - Provides vCPUs as sockets to the guest
* `cores` - Provides vCPUs as cores to the guest
* `threads` - Provides vCPUs as threads to the guest
* `spread` - Spreads vCPUs across sockets and cores by default. See the following [SpreadOptions](#### SpreadOptions) section for more details.

> 

Reviewer 2: Missing `any` option ?

Author: @RoniKishner  Good point, I could add a comment but `Any` isn't useful in the context of `preferredCPUTopology` as it just gets translated into `sockets` if provided with an instance type.

It's actually more useful in the context of using a preference without an instance type as it allows us to define a preference with a vCPU requirement that can be expressed as any combination of sockets, cores and threads within the VM.

This is a pretty bespoke use case that I didn't want to cover just now until the docs are broken up and we have more space to cover it.

Reviewer 2: A comment would do, some .yaml files might have it already and it would be confusing for them to see `any` and not know what it does.

Note that support for the original `preferSockets`, `preferCores`, `preferThreads` and `preferSpread` values for `PreferredCPUTopology` is deprecated as of `v1.4.0` ahead of removal in a future release.

#### SpreadOptions

When `spread` is provided as the value of `PreferredCPUTopology` we can further customize how vCPUs are spread across the guest visible CPU topology using `SpreadOptions`:

```yaml
apiVersion: instancetype.kubevirt.io/v1beta1
kind: VirtualMachinePreference
metadata:
  name: example-preference-cpu-topology
spec:
  cpu:
    preferredCPUTopology: spread
    spreadOptions:
      across: SocketsCoresThreads
      ratio: 4
```

[`spreadOptions`](https://kubevirt.io/api-reference/main/definitions.html#_v1beta1_spreadoptions) provides the following configurables:

* `across` - Defines how vCPUs should be spread across the guest visible CPU topology

> * `ratio`  - Defines the ratio at which vCPUs should be spread (defaults to 2)

Reviewer: Add note how this is supposed to mimic HT?

Author: As suggested below the ratio actually translates to the number of cores per socket with the three available `Across` values.  I can add some notes below for `SocketsCoresThreads` and `CoresThreads` highlighting that they require at least 2 vCPUs and mimic HT.
